app/experimenter/experiments/tasks.py

```diff
@@ -122,100 +121,124 @@ def update_experiment_bug_task(user_id, experiment_id):
 def update_experiment_info():
     metrics.incr("update_experiment_info.started")
     logger.info("Updating experiment info")
-    launch_experiments = Experiment.objects.filter(
-        Q(status=Experiment.STATUS_ACCEPTED) | Q(status=Experiment.STATUS_LIVE)
-    )
+    update_accepted_experiments()
+    update_live_experiments()
+    metrics.incr("update_experiment_info.completed")
+
 
-    for experiment in launch_experiments:
+def update_accepted_experiments():
+    accepted_experiments = Experiment.objects.filter(
+        status=Experiment.STATUS_ACCEPTED
+    )
+    for experiment in accepted_experiments:
         try:
             logger.info("Updating Experiment: {}".format(experiment))
             if experiment.normandy_id:
-                update_status(experiment)
-                if experiment.status == Experiment.STATUS_LIVE:
-                    send_period_ending_emails(experiment)
+                recipe_data = normandy.get_recipe(experiment.normandy_id)
+                if needs_to_be_updated(recipe_data, experiment.status):
+                    update_status(experiment, recipe_data)
```

Comment:
This structure is really good and clear, just do the accepted/live check before this